packages/open-api-gateway/src/construct/utils/api-gateway-integrations-types.ts

```diff
@@ -0,0 +1,84 @@
+/*********************************************************************************************************************
+ Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License").
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ******************************************************************************************************************** */
+
+import { AuthorizationType, CorsOptions } from "aws-cdk-lib/aws-apigateway";
+import { IFunction } from "aws-cdk-lib/aws-lambda";
+
+/**
+ * Defines an integration for an individual API operation
+ */
+export interface OpenApiIntegration {
+  /**
+   * The lambda function to service the api operation
+   */
+  readonly function: IFunction;
+}
+
+/**
+ * A mapping of operation id to the integration for that operation
+ */
+export type OpenApiIntegrations = {
+  readonly [operationId: string]: OpenApiIntegration;
+};
+
+/**
+ * Structure to contain an API operation's method and path
+ */
+export interface MethodAndPath {
+  /**
+   * The path of this operation in the api
+   */
+  readonly path: string;
+  /**
+   * The http method of this operation
+   */
+  readonly method: string;
```

Comment:
Good call, will define manually since this needs to be lowercase (type comes from OpenAPI and I can't reuse that since it's not compatible with jsii) :)